scipy/stats/_continuous_distns.py

```diff
@@ -2715,8 +2730,14 @@ def _stats(self, c):
         return mu, mu2, g1, g2
 
     def _entropy(self, c):
-        return -np.log(c) + sc.psi(c + 1) + _EULER + 1
-
+        return _lazywhere(c < 1e12, (c, ),
```

Comment:
As long as we're adding code to handle large `c`, we should try to get the most out of the asymptotic formula.  The following plot of the relative error of the basic formula and the asymptotic formula (based on using `mpmath` to compute the true value) shows that we can lower the threshold to about `8e6`.

![genlogistic_entropy_error](https://user-images.githubusercontent.com/321463/222310697-f9d34632-a5cb-4b8f-9630-56b82c644a49.png)

Here's the script that I used to generate that plot:

<details>
<summary>genlogistic entropy error plot</summary>

```
# genlogistic entropy work

import numpy as np
from scipy.special import psi
from mpmath import mp
import matplotlib.pyplot as plt


def basic(c):
    return -np.log(c) + psi(c+1) + np.euler_gamma + 1


def asymptotic(c):
    return 1/(2*c) + np.euler_gamma + 1


def mp_entropy(c, dps=100):
    with mp.workdps(dps):
        c = mp.mpf(c)
        return float(-mp.log(c) + mp.digamma(c + 1) + mp.euler + 1)


c = np.geomspace(4e6, 2e8, 10000)

ytrue = np.array([mp_entropy(t) for t in c])
y0 = basic(c)
yasym = asymptotic(c)


plt.plot(c, np.abs(y0 - ytrue)/ytrue, alpha=0.5, label='basic formula')
plt.plot(c, np.abs(yasym - ytrue)/ytrue, alpha=0.5, label='asymptotic')

plt.semilogx()
plt.grid()
plt.title('relative error, genlogistic entropy')
plt.xlabel('c')
plt.legend(framealpha=1, shadow=True)

plt.show()

```

</details>